### PR TITLE
[WIP] Enable WebSockets via Swoole

### PR DESCRIPTION
**Pull request by kingIZZZY**
## [WIP] Laravel Octane Swoole WebSockets

Already using Laravel Octane via Swoole?

Take full advantage of Swoole built-in WebSockets features to serve Laravel powered WebSockets!

Configure `octane.swoole.enableWebSockets` to `true` to enable

### CODE REVIEW

**Comment by kingIZZZY** [comment]
To access the Swoole server object from laravel code - here is one way how:
```php
app('Laravel\Octane\Swoole\WorkerState')->server
```
For example to push WebSocket messages to clients:
```php
app('Laravel\Octane\Swoole\WorkerState')->server->push($clientSocketFD, $message);
```

Is this the 1) only way? 2) correct way? If anyone knows - please share 🙏

File: src/Worker.php [hunk 1]
```diff
@@ -241,4 +243,44 @@ public function terminate(): void
     {
         $this->dispatchEvent($this->app, new WorkerStopping($this->app));
     }
+
+    /**
+     * Handle Incoming WebSocket Messages
+     */
+    public function handleWebSocketMessage(\Swoole\WebSocket\Server $server, \Swoole\WebSocket\Frame $frame)
+    {
+        \Laravel\Octane\CurrentApplication::set($sandbox = clone $this->app);
```
Comment (kingIZZZY) [inline src/Worker.php:252]: Pretty much copied from `handleTick`/`handleTask`, does it make sense? Should it be doing these same steps for incoming WebSocket messages/disconnect?

**Comment by taylorotwell** [comment]
Would need to be Echo compatible IMO.

**Comment by kingIZZZY** [comment]
Could we keep the PR open for a bit longer to 1) maybe hear some more opinions 2) maybe attract some more attention of others better well versed with Echo than myself who can take up contributing on that aspect?

Or is there an evident impossibility of compatibility with Echo which dooms this whole approach/effort that I am missing?

**Comment by goodevilgenius** [comment]
Personally, I think this looks like a really good start to me.

Only suggestion I have is that I don't think the union type for `Server|WServer` is necessary. Last time I checked, `Swoole\WebSocket\Server` extends `Swoole\Http\Server`. So, you can just type check the `Http` server and that covers both.

Making it work with echo would be the logical next step, though.

**Comment by kingIZZZY** [comment]
Oh no 😭 I think I might abandon this effort for now
It seems this feature is prone to [throwing this exception](https://github.com/laravel/octane/issues/906) possibly among others:
- BindingResolutionException: Target class [events] does not exist.

Because: Laravel does not support Swoole Coroutines
- https://github.com/laravel/octane/issues/765 specifically [this comment](https://github.com/laravel/octane/issues/765#issuecomment-2417130074)

Seems it might be a deal breaker for now, oh well - it's back to [Hyperf](https://github.com/hyperf/hyperf) for me...
Anyone who'd like can also check out the [Laravel-Hyperf](https://laravel-hyperf.com/) project (see the above linked issue #765 and specific comment)

**Comment by izdrail** [comment]
It's a nice project Hyperf. 
I like roardrunner better i don't know why

**Pull request closed without merging**